lib/membrane_rtc_engine/endpoints/webrtc/media_event.ex

```diff
@@ -266,22 +289,45 @@ defmodule Membrane.RTC.Engine.Endpoint.WebRTC.MediaEvent do
             "type" => "offer",
             "sdp" => sdp
           },
-          "trackIdToTrackMetadata" => track_id_to_track_metadata,
-          "midToTrackId" => mid_to_track_id
+          "trackIdToTrackInfo" => track_id_to_track_info
         }
       } ->
-        {:ok,
-         %{
-           type: :sdp_offer,
-           data: %{
-             sdp_offer: %{
-               type: :offer,
-               sdp: sdp
-             },
-             track_id_to_track_metadata: track_id_to_track_metadata,
-             mid_to_track_id: mid_to_track_id
-           }
-         }}
+        # max_bandwidth map still has rid strings as keys (e.g. %{"h" => 1500})
+        decoded_track_id_to_track_info =
+          track_id_to_track_info
+          |> Enum.map(fn {id, track_info} ->
+            {id,
+             %{
+               track_metadata: Map.get(track_info, "trackMetadata", :invalid),
+               max_bandwidth: Map.get(track_info, "maxBandwidth", :invalid),
+               mid: Map.get(track_info, "mid", :invalid)
+             }}
+          end)
+          |> Map.new()
+
+        is_valid =
+          for(
+            {_id, track_info} <- decoded_track_id_to_track_info,
+            {_key, value} <- track_info,
+            do: value != :invalid
+          )
+          |> Enum.all?()
```

Comment:
There must be some nicer way to check the correctness of this map, but I could not think of any.